### ISSUE TITLE
Add development docs for working with different versions

### ISSUE THREAD

**Issue by lea-hagen** [BEAST-Fitting/beast#564]
Now that we have a v2.0 and v1.4 (#510), I'm not sure about the procedure for working with the two branches.  I suspect I could find info on the internet, but since we already have some basic git instructions for doing development, it would be nice to add something about this too.

**Comment by karllark** [comment]
I think the procedure would be to have two conda environments, one for v1.4 and one for the dev version - v2.0.  Does this make sense?  This was the idea behind the "production" instructions.

**Comment by lea-hagen** [comment]
Yes, the conda environment part makes sense!  What I haven't figured out is how to check out the v1.4 branch, and then do any development on it.

**Comment by lea-hagen** [comment]
@astronomeralex and I found a method that perhaps works:
```
git fetch upstream
git checkout upstream/v1.4 -b v1.4
```
But when I do `python setup.py develop`, the output text says `Processing dependencies for beast==2.0.dev2222`.  Is that because v1.4 and v2.0 are still the same, or because something is processing incorrectly?

**Comment by karllark** [comment]
Apologies, I hadn't pushed the last few commits I did to the v1.4 branch for the v1.4 release.  Doh!  If you pull the v1.4 branch again, it should now report 1.4.1.dev is your base version.

**Comment by lea-hagen** [comment]
Yep, it shows `beast==1.4.1.dev2225` now!
